migration-tool/pkg/utils/utils.go

```diff
@@ -6,28 +6,23 @@ package utils
 
 import (
 	"fmt"
+	"sync"
 
 	"github.com/cespare/xxhash/v2"
 	"github.com/prometheus/prometheus/model/labels"
 	"github.com/prometheus/prometheus/prompb"
-	"github.com/timescale/promscale/pkg/clockcache"
 )
 
 const (
-	LabelJob           = "job"
-	Megabyte           = 1024 * 1024
-	maxLabelsCacheSize = 50 * 1000
+	LabelJob = "job"
+	Megabyte = 1024 * 1024
 )
 
 var (
-	labelsCache *clockcache.Cache
+	labelsCache sync.Map
 	seps        = []byte{'\xff'}
 )
 
-func init() {
-	labelsCache = clockcache.WithMax(maxLabelsCacheSize)
```

Comment:
Why are we changing labels cache in this PR?

Reply:
Ahh I see that we can't import Promscale. Iirc clockcache is sort of  LRU cache so this change can lead to different performance of prom-migrator? Maybe consider using some 3rd party LRU cache impl with limits in cache size to not run oom.

Reply:
The perf won't be a big deal, since now we are caching everything. Plus, a single-label entry is pretty light.

Reply:
We are currently limiting cache size to 50K and using LRU policy to evict, so it's much different from proposed change. I'd recommend going for LRU cache with the same size: eg https://github.com/hashicorp/golang-lru